src/main/java/fi/metatavu/metamind/server/rest/StoryExportApiImpl.java

```diff
@@ -0,0 +1,162 @@
+      List<UUID> intentMaterialIds = new ArrayList<UUID>();
+      for (int j = 0; j < intentMaterial.size(); j++) {
+        intentMaterialIds.add(intentMaterial.get(j).getId());
+      }
+      
+      exportedIntent.setTrainingMaterialIds(intentMaterialIds);
+      
+      storyToExportIntents.add(exportedIntent);
+    }
+    
+    exportedStory.setIntents(storyToExportIntents);
+    
+    List<TrainingMaterial> trainingMaterialsToExport = trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.INTENTOPENNLPDOCCAT, TrainingMaterialVisibility.LOCAL);
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.INTENTOPENNLPDOCCAT, TrainingMaterialVisibility.STORY));
+    
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.INTENTREGEX, TrainingMaterialVisibility.STORY));
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.INTENTREGEX, TrainingMaterialVisibility.LOCAL));
+    
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.VARIABLEOPENNLPNER, TrainingMaterialVisibility.STORY));
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.VARIABLEOPENNLPNER, TrainingMaterialVisibility.LOCAL));
+    
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.VARIABLEOPENNLPREGEX, TrainingMaterialVisibility.STORY));
+    trainingMaterialsToExport.addAll(trainingMaterialController.listTrainingMaterials(storyToExport, TrainingMaterialType.VARIABLEOPENNLPREGEX, TrainingMaterialVisibility.LOCAL));
+    
+    List<ExportedStoryTrainingMaterial> storyToExportTrainingMaterials = new ArrayList<ExportedStoryTrainingMaterial>();
+    
+    for (int i = 0; i < trainingMaterialsToExport.size(); i++) {
+      TrainingMaterial materialToExport = trainingMaterialsToExport.get(i);
+      ExportedStoryTrainingMaterial exportedMaterial = new ExportedStoryTrainingMaterial();
+      
+      exportedMaterial.setId(materialToExport.getId());
+      exportedMaterial.setType(materialToExport.getType());
+      exportedMaterial.setText(materialToExport.getText());
+      exportedMaterial.setVisibility(materialToExport.getVisibility().toString());
+      exportedMaterial.setName(materialToExport.getName());
+      
+      storyToExportTrainingMaterials.add(exportedMaterial);
+    }
+    
+    exportedStory.setTrainingMaterials(storyToExportTrainingMaterials);
+    
+    List<Variable> variablesToExport = storyController.listVariablesByStory(storyToExport);
+    List<ExportedStoryVariable> exportedStoryVariables = new ArrayList<ExportedStoryVariable>();
+    
+    for (int i = 0; i < variablesToExport.size(); i++) {
```

Comment:
Could also use streams and map function eq.
```suggestion
exportedStory.setVariables(variablesToExport
  .stream()
  .map((var) -> {
      ExportedStoryVariable exportedVariable = new ExportedStoryVariable();
      exportedVariable.setName(var.getName());
      exportedVariable.setType(var.getType());
      exportedVariable.setValidationScript(var.getValidationScript());
  }).collect(Collectors.toList()));
```